app-emulation/crun: bump to 0.16

robertgzr opened the pull request:
Signed-off-by: Robert Günzler <r@gnzler.io>

robertgzr commented:
@juippis let me know when it's time to drop older version... I was thinking of taking 0.14 out or at least replacing 0.15 with 0.15.1

juippis commented:
If it were me, I'd just have 0.16 in tree and delete everything else. If there's nothing depending on 0.15 series, people will install 0.16 by default anyway since they all have matching KEYWORDS.

So can we not at least add 0.15.1?

juippis commented on app-emulation/crun/crun-0.16.ebuild:
```diff
@@ -0,0 +1,68 @@
+		$(use_enable criu) \
+		$(use_enable seccomp) \
+		$(use_enable systemd) \
+		$(usex static-libs '--enabled-shared  --enabled-static' '--enable-shared --disable-static' '' '')
```
This doesn't seem to work, when `+static-libs`:
```
 * econf: updating crun-0.16/build-aux/config.guess with /usr/share/gnuconfig/config.guess
./configure --prefix=/usr --build=x86_64-pc-linux-gnu --host=x86_64-pc-linux-gnu --mandir=/usr/share/man --infodir=/usr/share/info --datadir=/usr/share --sysconfdir=/etc --localstatedir=/var/lib --disable-dependency-tracking --disable-silent-rules --docdir=/usr/share/doc/crun-0.16 --htmldir=/usr/share/doc/crun-0.16/html --with-sysroot=/ --libdir=/usr/lib64 --enable-bpf --disable-caps --enable-criu --disable-seccomp --disable-systemd --enabled-shared --enabled-static
configure: error: unrecognized option: `--enabled-shared'
Try `./configure --help' for more information
 * ERROR: app-emulation/crun-0.16::testworld failed (configure phase):
 *   econf failed
 * 
```

juippis reviewed (commented):
Also don't push 0.15.1, as it has same KEYWORDS as 0.16 so probably no one will end up installing it. Rather just leave the 0.15 in tree for people to go back if they can't compile 0.16.

robertgzr commented:
alright, sorry for the long silence...

- kept 0.15 unchanged
- dropped 0.14.1
- introduced 0.16
  - with the static-build fixes and
  - dropping the `man` use flag (thanks @epsilon-0)

juippis commented on app-emulation/crun/crun-0.16.ebuild:
```diff
@@ -0,0 +1,61 @@
+LICENSE="GPL-2+ LGPL-2.1+"
+SLOT="0"
+KEYWORDS="~amd64 ~arm ~arm64 ~ppc64"
+IUSE="+bpf +caps criu +seccomp systemd static-libs"
```
Btw we are trying to get rid of `static-libs` where they're not needed.

robertgzr replied:
what's the reason? I always thought it a strong point of gentoo to be able to get static libraries easily when I need them...

juippis replied:
https://projects.gentoo.org/qa/policy-guide/installed-files.html#pg0302

juippis commented on app-emulation/crun/crun-0.16.ebuild:
```diff
@@ -0,0 +1,61 @@
+		$(use_enable criu) \
+		$(use_enable seccomp) \
+		$(use_enable systemd) \
+		$(usex static-libs '--enable-shared --enable-static' '--enable-shared --disable-static' '' '')
```
Backlashes not needed with array, one of the arrays strong points. Will clean after you.

robertgzr replied:
thanks o/

juippis reviewed (approved):
```
/usr/lib/gcc/x86_64-pc-linux-gnu/10.2.0/../../../../x86_64-pc-linux-gnu/bin/ld: ./.libs/libcrun.so: undefined reference to `seccomp_init'
/usr/lib/gcc/x86_64-pc-linux-gnu/10.2.0/../../../../x86_64-pc-linux-gnu/bin/ld: ./.libs/libcrun.so: undefined reference to `seccomp_rule_add'
/usr/lib/gcc/x86_64-pc-linux-gnu/10.2.0/../../../../x86_64-pc-linux-gnu/bin/ld: ./.libs/libcrun.so: undefined reference to `seccomp_arch_add'
/usr/lib/gcc/x86_64-pc-linux-gnu/10.2.0/../../../../x86_64-pc-linux-gnu/bin/ld: ./.libs/libcrun.so: undefined reference to `seccomp_export_bpf'
/usr/lib/gcc/x86_64-pc-linux-gnu/10.2.0/../../../../x86_64-pc-linux-gnu/bin/ld: ./.libs/libcrun.so: undefined reference to `seccomp_release'
/usr/lib/gcc/x86_64-pc-linux-gnu/10.2.0/../../../../x86_64-pc-linux-gnu/bin/ld: ./.libs/libcrun.so: undefined reference to `seccomp_rule_add_array'
/usr/lib/gcc/x86_64-pc-linux-gnu/10.2.0/../../../../x86_64-pc-linux-gnu/bin/ld: ./.libs/libcrun.so: undefined reference to `seccomp_syscall_resolve_name'
collect2: error: ld returned 1 exit status
make: *** [Makefile:1143: crun] Error 1
 * ERROR: app-emulation/crun-0.16::testworld failed (compile phase):
 *   emake failed
```
Is this just my system? This doesn't seem to emerge with `-seccomp` USE at all.
